src/Enterspeed.Source.UmbracoCms/HostedServices/HandleEnterspeedJobsHostedService.cs

```diff
@@ -48,13 +48,13 @@ public override Task PerformExecuteAsync(object state)
                     return Task.CompletedTask;
                 }
 
-                if (serverRoleAccessor.CurrentServerRole == ServerRole.SchedulingPublisher || serverRoleAccessor.CurrentServerRole == ServerRole.Single)
+                if(configurationService.RunJobsOnServer(serverRoleAccessor.CurrentServerRole))
```

Comment:
I was curious to a different approach for this, since we use it quite a few places, and depend on the serverRoleAccessor in the classes that uses this method. 

Would it be an idea to do something like IServerRole.CanRunJobs() or IJobManager/Handler.CanRun(), where the implementation contains the configurationService as well as serverRoleAccessor? 

I am not sure if it is super important, it was just a thought of removing some dependencies on the implementing classes.

Reply:
I have updated the implementation, please take a look again and see if it better.